src/components/FilterPanel.jsx

```diff
@@ -18,38 +20,66 @@ import CloseIcon from '@mui/icons-material/Close';
 import PropTypes from 'prop-types';
 
 export default function FilterPanel({ onClose }) {
-  const [{ minAmount, maxAmount, grades, majors }, setQueryParam] =
+  const [{ minAmount, maxAmount, grades, majors }, setQueryParams] =
     useQueryParams();
+  const [filterVals, setFilterVals] = useState({
```

Comment:
How about using useState for each one of these and calling what useQueryParams returns just `params` instead of the individual filter names?

Reply:
so I would change `setQueryParam` to how it originally was and then call it on each seperate fitler?

Reply:
You can call `setQueryParams({minAmount, majors, etc.})`